make-bm-worker/README.md

```diff
@@ -0,0 +1,68 @@
+Registering Bare Metal Hosts
+============================
+
+The `make-bm-worker` tool may be a more convenient way of creating
+YAML definitions for workers than editing the files directly.
+
+```
+$ go run make-bm-worker/main.go -address 1.2.3.4 -password password -user admin worker-99
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: worker-99-bmc-secret
+type: Opaque
+data:
+  username: YWRtaW4=
+  password: cGFzc3dvcmQ=
+
+---
+apiVersion: metalkube.org/v1alpha1
+kind: BareMetalHost
+metadata:
+  name: worker-99
+spec:
+  online: true
+  bmc:
+    address: 1.2.3.4
+    credentialsName: worker-99-bmc-secret
+```
+
+The output can be passed directly to `oc apply` like this:
+
+```
+$ go run make-bm-worker/main.go -address 1.2.3.4 -password password -user admin worker-99 | oc apply -f -
+```
+
+Include the `-image` option to include the image settings needed to
+trigger immediate provisioning:
+
+```
+$ go run make-bm-worker/main.go -address 1.2.3.4 -password password -user admin -image worker-99
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: worker-99-bmc-secret
+type: Opaque
+data:
+  username: YWRtaW4=
+  password: cGFzc3dvcmQ=
+
+---
+apiVersion: metalkube.org/v1alpha1
+kind: BareMetalHost
+metadata:
+  name: worker-99
+spec:
+  online: true
+  bmc:
+    address: 1.2.3.4
+    credentialsName: worker-99-bmc-secret
+
+  userData:
+    namespace: openshift-machine-api
+    name: worker-user-data
+  image:
+    url: "ERROR: template: yaml_out:26:13: executing "yaml_out" at <.ImageSourceURL>: can't evaluate field ImageSourceURL in type main.TemplateArgs
```

Comment:
What should the url be when this isn't an error - do we pass the URL for the webserver on the host for now, until we figure out where to host the image on the cluster?

Reply:
Tested and it works fine, so I guess we just need to fix the README?

```
    url: "http://172.22.0.1/images/redhat-coreos-maipo-latest.qcow2"
    checksum: "http://172.22.0.1/images/redhat-coreos-maipo-latest.qcow2.md5sum"
```

Reply:
D'oh, I didn't notice that error message. I've updated it with a real value.

The actual URL to use will depend on the resolution to the question of how we're going to serve that image once the cluster is online. As long as the server on the provisioning host is available, the value there now will work.